### PR TITLE
263 youtube title parsing

### PR DESCRIPTION
**Pull request by jottinger**

### CODE REVIEW

File: src/main/kotlin/javabot/operations/BrowseOperation.kt [hunk 1]
```diff
@@ -63,18 +55,37 @@ class BrowseOperation @Inject constructor(bot: Javabot, adminDao: AdminDao,
                             "and module `$module` "
                         } else {
                             ""
-                        }
+                        } +
+                        "found."
             }
         } catch (e: Throwable) {
-            "ERR: not found"
+            throw e
```
Comment (evanchooly) [inline src/main/kotlin/javabot/operations/BrowseOperation.kt]: just lose the try/catch?

File: src/main/kotlin/javabot/operations/KarmaOperation.kt [hunk 2]
```diff
@@ -59,15 +59,25 @@ class KarmaOperation @Inject constructor(bot: Javabot, adminDao: AdminDao, var d
...
-                log.info("message = " + message, e)
-                throw e
-            }
+            val nick: String =
+                    try {
+                        val temp = message.substring(0, operationPointer).trim().toLowerCase()
+                        // got an empty nick; spaces only?
+                        if (temp.isEmpty()) {
+                            // need to check for special case where bot was *addressed* for karma
+                            if (event.addressed == true) {
```
Comment (evanchooly) [inline src/main/kotlin/javabot/operations/KarmaOperation.kt]: == true?  really?  :)

File: src/main/kotlin/javabot/operations/URLTitleOperation.kt [hunk 3]
```diff
@@ -44,8 +49,8 @@ class URLTitleOperation @Inject constructor(bot: Javabot, adminDao: AdminDao,
         val title = if (titlesToPost.size == 1) "title" else "titles"
         responses.add(Message(event, "${event.user.nick}'s $title: " +
                 titlesToPost.joinToString(" | ", transform = { s ->
-                    if (s.length > 250) {
-                        "\"${s.substring(0, 260)}...}\""
+                    if (s.length > 240) {
+                        "\"${s.substring(0, 240)}...\"\""
```
Comment (evanchooly) [inline src/main/kotlin/javabot/operations/URLTitleOperation.kt]: maybe use """ so you don't need to escape

Comment (jottinger) [reply src/main/kotlin/javabot/operations/URLTitleOperation.kt]: That's uuuuugly but doable (and done)

**Pull request merged**